pkg/externalLinks/ExternalLinksService.go

```diff
@@ -0,0 +1,311 @@
+	if err0 != nil {
+		impl.logger.Errorw("No matching entry found for update.", "id", request.Id)
+		return nil, err0
+	}
+	externalLinks.Name = request.Name
+	externalLinks.Url = request.Url
+	externalLinks.Active = true
+	externalLinks.ExternalLinksMonitoringToolId = request.MonitoringToolId
+	externalLinks.UpdatedBy = int32(request.UserId)
+	externalLinks.UpdatedOn = time.Now()
+	err := impl.externalLinksRepository.Update(&externalLinks)
+	if err != nil {
+		impl.logger.Errorw("error in updating link", "data", externalLinks, "err", err)
+		return nil, err
+	}
+
+	allExternalLinksMapping, err := impl.externalLinksClustersRepository.FindAllByExternalLinkId(request.Id)
+	if err != nil {
+		impl.logger.Errorw("error in fetching link", "data", externalLinks, "err", err)
+		return nil, err
+	}
+	for _, model := range allExternalLinksMapping {
+		model.Active = false
+		model.UpdatedBy = int32(request.UserId)
+		model.UpdatedOn = time.Now()
+		err := impl.externalLinksClustersRepository.Update(model)
+		if err != nil {
+			impl.logger.Errorw("error in updating clusters to false", "data", model, "err", err)
+			return nil, err
+		}
+	}
+	for _, requestedClusterId := range request.ClusterIds {
+		externalLinkClusterId := 0
+		var externalLinkCluster *ExternalLinksClusters
+		for _, model := range allExternalLinksMapping {
+			if requestedClusterId == model.ClusterId {
+				externalLinkClusterId = model.Id
+				externalLinkCluster = model
+				break
+			}
+		}
+		if externalLinkClusterId > 0 && externalLinkCluster != nil {
+			externalLinkCluster.Active = true
+			externalLinkCluster.UpdatedOn = time.Now()
+			externalLinkCluster.UpdatedBy = request.UserId
+			err = impl.externalLinksClustersRepository.Update(externalLinkCluster)
+		} else {
+			externalLinkCluster := &ExternalLinksClusters{
+				ExternalLinksId: request.Id,
+				ClusterId:       requestedClusterId,
+				Active:          true,
+				AuditLog:        sql.AuditLog{CreatedOn: time.Now(), CreatedBy: request.UserId, UpdatedOn: time.Now(), UpdatedBy: request.UserId},
+			}
+			err = impl.externalLinksClustersRepository.Save(externalLinkCluster)
+		}
+		if err != nil {
+			impl.logger.Errorw("error in saving cluster id's", "data", externalLinkCluster, "err", err)
+			err = &util.ApiError{
+				InternalMessage: "cluster id failed to create in db",
+				UserMessage:     "cluster id failed to create in db",
+			}
+			return nil, err
+		}
+	}
+	externalLinksCreateUpdateResponse := &ExternalLinksCreateUpdateResponse{
+		Success: true,
+	}
+	return externalLinksCreateUpdateResponse, nil
+}
+func (impl ExternalLinksServiceImpl) DeleteLink(id int, userId int32) (*ExternalLinksCreateUpdateResponse, error) {
+	impl.logger.Debugw("link delete request", "req", id)
+	externalLinksMapping, err := impl.externalLinksClustersRepository.FindAllActiveByExternalLinkId(id)
+	if err != nil {
+		return nil, err
+	}
+	for _, externalLink := range externalLinksMapping {
+		externalLink.Active = false
```

Comment:
Do we need to update in clusterMapping table as well?
Only updation in external_link table won't be sufficient?

Reply:
we have to update its mapping every time when links are attached/ de-attached on request.